src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -18693,6 +18816,50 @@ public void Main()
             VerifyAnnotationsAndMetadata(c, "C.M", EnsuresNotNull);
         }
 
+        [Fact, WorkItem(32172, "https://github.com/dotnet/roslyn/issues/32172")]
+        public void NotNullWhenTrue_WithNotNullWhenFalse_UserDefinedOperator()
+        {
+            // When both NotNullWhenTrue and NotNullWhenFalse are applied, it's the same as EnsuresNotNull,
+            // even if the method doesn't return bool.
+            CSharpCompilation c = CreateCompilation(new[] { @"
+#pragma warning disable CS0660, CS0661 // no equals, hashcode 
+using System.Runtime.CompilerServices;
+public class C
+{
+    public void Main(C? c, C? c2)
+    {
+        if(c == c2)
+        {
+         	c.ToString();   
+        }
+        else
+        {
+            c.ToString(); 
+        }
+
+        if (c != c2)
+        {
+            c2.ToString();
+        }
+    }
```

Comment:
>} [](start = 4, length = 1)

For completeness, consider calling `c.ToString(); c2.ToString();` in each branch. (At that point, it would be better to have two methods, one testing `==` and the other testing `!=`, so there's no side-effects from earlier calls to `ToString()`.)